Source/Utilis/Protos/GenericMessage+Content.swift

```diff
@@ -22,192 +22,102 @@ import Foundation
 
 public extension GenericMessage {
     var hasText: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .text:
-            return true
-        default:
-            return false
+        guard let _ = messageData as? Text else {
+            let ephemeral = messageData as? Ephemeral
+            return ephemeral?.hasText ?? false
         }
+        return true
     }
     
     var hasConfirmation: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .confirmation:
-            return true
-        default:
-            return false
-        }
+        return (messageData as? Confirmation) != nil ? true : false
```

Comment:
```suggestion
        return messageData is Confirmation
```

I guess we can do the same for all other cases as well. Maybe we can even remove the hasXXX methods completely but I guess that will affect a lot of tests so feel free to keep them.